griptape/drivers/vector/pgvector_vector_store_driver.py

```diff
@@ -0,0 +1,159 @@
+        """
+        self.engine.execute('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";')
+        self.engine.execute('CREATE EXTENSION IF NOT EXISTS "vector";')
+
+    def create_schema(self) -> None:
+        """
+        Creates the schema for the vectors table.
+        """
+        self.VectorModel.metadata.create_all(self.engine)
+
+    def upsert_vector(
+        self,
+        vector: list[float],
+        vector_id: Optional[str] = None,
+        namespace: Optional[str] = None,
+        meta: Optional[dict] = None,
+        **kwargs
+    ) -> str:
+        """
+        Inserts or updates a vector in the collection.
+        """
+        with Session(self.engine) as session:
+            obj = self.VectorModel(
+                id=vector_id,
+                vector=vector,
+                namespace=namespace,
+                meta=meta,
+            )
+
+            obj = session.merge(obj)
+            session.commit()
+
+            return str(obj.id)
+
+    def load_entry(self, vector_id: str, namespace: Optional[str] = None) -> BaseVectorStoreDriver.Entry:
+        """
+        Retrieves a specific vector entry from the collection based on its identifier and optional namespace.
+        """
+        with Session(self.engine) as session:
+            result = session.get(self.VectorModel, vector_id)
+
+            return BaseVectorStoreDriver.Entry(
+                id=result.id,
+                vector=result.vector,
+                namespace=result.namespace,
+                meta=result.meta,
+            )
+
+    def load_entries(self, namespace: Optional[str] = None) -> list[BaseVectorStoreDriver.Entry]:
+        """
+        Retrieves all vector entries from the collection, optionally filtering to only
+        those that match the provided namespace.
+        """
+        with Session(self.engine) as session:
+            query = session.query(self.VectorModel)
+            if namespace:
+                query = query.filter_by(namespace=namespace)
+
+            results = query.all()
+
+            return [
+                BaseVectorStoreDriver.Entry(
+                    id=str(result.id),
+                    vector=result.vector,
+                    namespace=result.namespace,
+                    meta=result.meta,
+                )
+                for result in results
+            ]
+
+    def query(
+        self,
+        query: str,
+        count: Optional[int] = None,
+        namespace: Optional[str] = None,
+        include_vectors: bool = False,
+        distance_metric: str = "cosine_distance",
+        **kwargs
+    ) -> list[BaseVectorStoreDriver.QueryResult]:
+        """
+        Performs a search on the collection to find vectors similar to the provided input vector,
+        optionally filtering to only those that match the provided namespace.
+        """
+        count = count if count else BaseVectorStoreDriver.DEFAULT_QUERY_COUNT
```

Comment:
Can we move this to the parameter default? Instead of count defaulting to `None` have it default to `BaseVectorStoreDriver.DEFAULT_QUERY_COUNT`